types/enzyme-adapter-react-16/index.d.ts

```diff
@@ -12,4 +12,4 @@ declare class ReactSixteenAdapter extends EnzymeAdapter {
 declare namespace ReactSixteenAdapter {
 }
 
-export = ReactSixteenAdapter;
+export default ReactSixteenAdapter;
```

Comment:
Are we sure this is correct? The actual JS uses `module.exports = `. TypeScript's ES module interop support, which would allow this to work, was only added in [TypeScript 2.7](www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#support-for-import-d-from-cjs-form-commonjs-modules-with---esmoduleinterop) behind the `--esModuleInterop` flag. It is not the default behavior, and is definitely not supported in TypeScript 2.6 or earlier.

Reply:
Hm, the test passes with 2.6 for me. I've also created a very basic 2.6 project where this type of export also works: https://github.com/saenglert/tstest

Reply:
Its not the test passing, but actual runtime evaluation. Since the actual JS file for `enzyme-adapter-react-16` uses `module.exports =`, this is not an actual `default` export. 

TypeScript 2.7's `--esModuleInterop` flag adds an extra helper that converts the imports of a module using `module.exports=` into a module object with a synthetic `default`, but only in that version of TS and later and with that flag set.

Also, your example project isn't demonstrating the issue. If you use `export default` in one file and `import x from` in another, that works fine because TypeScript emits the default as `exports.default = `, not `module.exports =`.

This section of the DT README has more information about why this change is not recommended: https://github.com/DefinitelyTyped/DefinitelyTyped#should-i-add-an-empty-namespace-to-a-package-that-doesnt-export-a-module-to-use-es6-style-imports